src/main/java/org/opensearch/security/securityconf/impl/v7/RoleV7.java

```diff
@@ -52,28 +70,36 @@ public static RoleV7 fromYamlString(String yamlString) throws IOException {
      * useful for tests.
      */
     public static RoleV7 fromYamlStringUnchecked(String yamlString) {
-        try {
-            return fromYamlString(yamlString);
+        try (Reader yamlReader = new StringReader(yamlString)) {
+            return fromYaml(yamlReader);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    public static RoleV7 fromYaml(URL yamlFile) throws IOException {
+        try (InputStream in = yamlFile.openStream(); Reader yamlReader = new InputStreamReader(in, StandardCharsets.UTF_8)) {
+            return fromYaml(yamlReader);
+        }
+    }
+
     public static RoleV7 fromYaml(Reader yamlReader) throws IOException {
         return DefaultObjectMapper.YAML_MAPPER.readValue(yamlReader, RoleV7.class);
     }
 
-    private boolean reserved;
-    private boolean hidden;
-    @JsonProperty(value = "static")
-    private boolean _static;
-    private String description;
-    private List<String> cluster_permissions = Collections.emptyList();
-    private List<Index> index_permissions = Collections.emptyList();
-    private List<Tenant> tenant_permissions = Collections.emptyList();
+    /**
+     * Does additional validations regarding limitiations of plugin permissions files.
+     */
+    public static RoleV7 fromPluginPermissionsFile(URL pluginPermissionsFile) throws IOException {
+        RoleV7 role = fromYaml(pluginPermissionsFile);
 
-    public RoleV7() {
+        if (role.tenant_permissions != null && !role.tenant_permissions.isEmpty()) {
```

Comment:
nit: we should probly check that only two keys exist instead of checking whether tenant_permissions is populated.

Reply:
It is effectively the same as there are only the 3 keys supported and I don't foresee adding another section..if anything `tenant_permissions` may be phased out in the future.